apps/wallet/src/ui/app/hooks/useLedgerNotification.tsx

```diff
@@ -0,0 +1,115 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { useFeatureIsOn } from '@growthbook/growthbook-react';
+import { X12 } from '@mysten/icons';
+import { useEffect } from 'react';
+import toast from 'react-hot-toast';
+import { useNavigate } from 'react-router-dom';
+import Browser from 'webextension-polyfill';
+
+import { trackEvent } from '../../../shared/plausible';
+import { useMenuIsOpen, useNextMenuUrl } from '../components/menu/hooks';
+import { AppType } from '../redux/slices/app/AppType';
+import { ButtonOrLink } from '../shared/utils/ButtonOrLink';
+import useAppSelector from './useAppSelector';
+
+const HAS_ACKNOWLEDGED_LEDGER_NOTIFICATION_KEY =
+    'has-acknowledged-ledger-notification';
+const HAS_ACKNOWLEDGED_LEDGER_NOTIFICATION_VALUE = 'true';
+
+const LEDGER_NOTIFICATION_TOAST_ID = 'ledger-notification-toast';
+
+// TODO: Delete this *soon* because custom, imperative toasts shouldn't be used for notifications :)
+export function useLedgerNotification() {
+    const isLedgerNotificationEnabled = useFeatureIsOn(
+        'wallet-ledger-notification-enabled'
+    );
+    const isMenuOpen = useMenuIsOpen();
+    const appType = useAppSelector((state) => state.app.appType);
+    const navigate = useNavigate();
+    const connectLedgerModalUrl = useNextMenuUrl(
+        true,
+        '/accounts/connect-ledger-modal'
+    );
+
+    useEffect(() => {
+        if (isMenuOpen) {
+            toast.remove(LEDGER_NOTIFICATION_TOAST_ID);
+        }
+    }, [isMenuOpen]);
+
+    useEffect(() => {
+        const hasAcknowledgedVal = localStorage.getItem(
+            HAS_ACKNOWLEDGED_LEDGER_NOTIFICATION_KEY
+        );
+        const hasAcknowledged =
+            hasAcknowledgedVal === HAS_ACKNOWLEDGED_LEDGER_NOTIFICATION_VALUE;
+
+        if (isLedgerNotificationEnabled && !hasAcknowledged && !isMenuOpen) {
```

Comment:
we only want to show the toast when the feature flag is on, the user hasn't seen it before, and the menu isn't actively opened